pennylane/ops/op_math/condition.py

```diff
@@ -112,6 +114,16 @@ def cond(condition, true_fn, false_fn=None, elifs=()):
     will be captured by Catalyst, the just-in-time (JIT) compiler, with the executed
     branch determined at runtime. For more details, please see :func:`catalyst.cond`.
 
+    When used with :func:`~.pennylane.capture.enabled`, this function allows for general
+    if-elif-else constructs. As with the JIT mode, all branches are captured,
+    with the executed branch determined at runtime.
+    However, the function cannot branch on mid-circuit measurements.

```

Comment:
Not sure this is worth writing out.  @mudit2812 's work on mcm's will make it so that measurement values are captured as booleans and compatible with this PR. So we will be able to integrate the two together.